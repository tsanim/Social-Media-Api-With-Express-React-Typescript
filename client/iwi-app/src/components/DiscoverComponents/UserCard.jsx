import React from 'react';
import URI from '../../config/config';
import PropTypes from 'prop-types';

function UserCard(props) {
    const { imageId, username, firstName, lastName } = props.user;

    return (
            <div className="card">
                <figure>
                    <img src={`${URI}/feed/image/${imageId}`} alt="" />
                    <div className="names">
                        <figcaption>{username}</figcaption>
                        <span className="fullname">{firstName} {lastName}</span>
                    </div>
                </figure>
            </div>
    )
}

UserCard.propTypes = {
    user: PropTypes.object,
}

export default UserCard;