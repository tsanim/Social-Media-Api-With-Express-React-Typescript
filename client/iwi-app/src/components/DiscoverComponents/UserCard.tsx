import React from 'react';
import URI from '../../config/config';
import PropTypes from 'prop-types';
import UserCardProps from '../../interfaces/Components/Discover/UserCardProps.interface';

function UserCard(props: UserCardProps) {
    const { imageId, username, firstName, lastName } = props.user;

    return (
            <div className="card">
                <figure>
                    <img src={`${URI}/feed/image/${imageId}`} alt="" />
                    <div className="names">
                        <figcaption>{username}</figcaption>
                        <span className="fullname">{firstName} {lastName}</span>
                    </div>
                </figure>
            </div>
    )
}

UserCard.propTypes = {
    user: PropTypes.object,
}

export default UserCard;