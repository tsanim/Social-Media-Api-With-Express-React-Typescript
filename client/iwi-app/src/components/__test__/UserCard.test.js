import React from 'react';

import { shallow } from 'enzyme';
import UserCard from '../DiscoverComponents/UserCard';

function userCardComponentSetup() {
    const props = {
        user: {},
    }

    const enzymeWrapper = shallow(<UserCard {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('UsesCard component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = userCardComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render div tag with class "card"', () => {
        const { enzymeWrapper } = userCardComponentSetup();

        expect(enzymeWrapper.find('div.card')).toHaveLength(1);
    });

    it('should render a figure tag', () => {
        const { enzymeWrapper } = userCardComponentSetup();

        expect(enzymeWrapper.find('figure')).toHaveLength(1);
    });

    it('should render img tag', () => {
        const { enzymeWrapper } = userCardComponentSetup();

        expect(enzymeWrapper.find('img')).toHaveLength(1);
    });

    it('should render div tag with class "names"', () => {
        const { enzymeWrapper } = userCardComponentSetup();

        expect(enzymeWrapper.find('div.names')).toHaveLength(1);
    });

    it('should render figcaption tag', () => {
        const { enzymeWrapper } = userCardComponentSetup();

        expect(enzymeWrapper.find('figcaption')).toHaveLength(1);
    });

    it('should render span tag with class "fullname"', () => {
        const { enzymeWrapper } = userCardComponentSetup();

        expect(enzymeWrapper.find('span.fullname')).toHaveLength(1);
    });
});