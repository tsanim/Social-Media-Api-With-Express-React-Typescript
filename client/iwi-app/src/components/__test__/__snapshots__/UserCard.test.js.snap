// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsesCard component should render initial layout 1`] = `
Array [
  <div
    className="card"
  >
    <figure>
      <img
        alt=""
        src="http://localhost:9999/feed/image/undefined"
      />
      <div
        className="names"
      >
        <figcaption />
        <span
          className="fullname"
        >
           
        </span>
      </div>
    </figure>
  </div>,
]
`;
